pkg/kubelet/cm/qos_container_manager_linux.go

```diff
@@ -106,9 +113,90 @@ func (m *qosContainerManagerImpl) Start(nodeInfo *v1.Node, activePods ActivePods
 	m.nodeInfo = nodeInfo
 	m.activePods = activePods
 
+	// TODO: We already have hooks in killPod() and syncPod() that
+	// update the QoS reserve on pod entry/exit. This is really only
+	// needed in the case that one of the QoS cgroups is over its
+	// resoruce limit and we are wanting to put continuous downward
+	// pressure on the cgroup such that we can eventually get to the
+	// desired limit.
+	go wait.Until(func() {
+		err := m.UpdateCgroups()
+		if err != nil {
+			glog.Warningf("[ContainerManager] Failed to reserve QoS requests: %v", err)
+		}
+	}, 30*time.Second, wait.NeverStop)
+
 	return nil
 }
 
+// setMemoryReserve sums the memory limits of all pods in a QOS class,
+// calculates QOS class memory limits, and set those limits in the
+// CgroupConfig for each QOS class.
+func (m *qosContainerManagerImpl) setMemoryReserve(configs map[v1.PodQOSClass]*CgroupConfig, percentReserve int64) {
+	qosMemoryRequests := map[v1.PodQOSClass]int64{
+		v1.PodQOSGuaranteed: 0,
+		v1.PodQOSBurstable:  0,
+	}
+
+	// Sum the pod limits for pods in each QOS class
+	pods := m.activePods()
+	for _, pod := range pods {
+		podMemoryRequest := int64(0)
+		qosClass := qos.GetPodQOS(pod)
+		if qosClass == v1.PodQOSBestEffort {
+			// limits are not set for Best Effort pods
+			continue
+		}
+		for _, container := range pod.Spec.Containers {
+			podMemoryRequest += container.Resources.Requests.Memory().Value()
```

Comment:
@sjenning -- you appeared to miss this comment, while what you have is functionally correct, the helper is consistent w/ what we are doing on cpu requests...